src/qtgui/bookmarks.h

```diff
@@ -30,26 +30,27 @@
 #include <QList>
 #include <QStringList>
 #include <QColor>
+#include <memory>
 
 struct TagInfo
 {
+    using sptr = std::shared_ptr<TagInfo>;
     QString name;
     QColor color;
     bool active;
 
     static const QColor DefaultColor;
     static const QString strUntagged;
 
-    TagInfo()
+    TagInfo(QString name = "")
```

Comment:
May-be simplify this by by #1176 first (if it makes sense to you, can you help me to get the MR in?)

But definitely a blocker, I can easily rebase. (I would suggest not to merge the change from #1176 to avoid mixing thinks together).

Reply:
#1176 looks fine to me. Approved. I'll rebase this PR if #1176 gets merged first.